presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1126,6 +1129,28 @@ protected Scope visitJoin(Join node, Optional<Scope> scope)
                 return output;
             }
 
+            if (criteria instanceof JoinUsing) {
```

Comment:
why not to make this an `else` block of `if` in 1122 line? I am afraid this code might be left when legacy flag is removed